lib/uplinkc/uplink_definitions.h

```diff
@@ -31,6 +31,19 @@ typedef struct Object           { long _handle; } ObjectRef;
 typedef struct Downloader       { long _handle; } DownloaderRef;
 typedef struct Uploader         { long _handle; } UploaderRef;
 typedef struct EncryptionAccess { long _handle; } EncryptionAccessRef;
+typedef struct CaveatPaths      { long _handle; } CaveatPathsRef;
+typedef struct CaveatNonce      { long _handle; } CaveatNonceRef;
+
+typedef struct Caveat {
+    bool disallow_reads;
+    bool disallow_writes;
+    bool disallow_lists;
+    bool disallow_deletes;
+    CaveatPathsRef allowed_paths;
+    int64_t not_after;
+    int64_t not_before;
+    CaveatNonceRef nonce;
```

Comment:
Having this as a ref feels weird.

Reply:
Maybe this should be `uint8_t *nonce; uint32_t nonce_length;`?

Reply:
See my comment above - I'm not sure how to solve this exactly.